github-get-reviewers.go

```diff
@@ -33,38 +29,76 @@ type ReviewState struct {
 	State string `json:"state"`
 }
 
-var rStatus []ReviewStatus
-var rReviews []ReviewState
+func getReviewStatesForPR(repo string, number int) (error, []ReviewState) {
+	reviews := make(map[string][]github.PullRequestReview)
 
-func getReviewers(repo string, number int) []ReviewStatus {
-	rStatus = nil
 	// List all reviewers to an assigned PR # in a repo.
-	rList, _, err := buzzClient.PullRequests.ListReviewers(ctx, "minio", repo, number, nil)
+	reviewers, _, err := buzzClient.PullRequests.ListReviewers(ctx, "minio", repo, number, nil)
 	if err != nil {
-		// We should not exit on error in this case.
-		fmt.Println("Unable to get Reviewers for %s", number)
+		return err, nil
 	}
-	for _, elem := range rList {
-		eachReviewer := ReviewStatus{}
-		eachReviewer.User.Login = elem.GetLogin()
-		rStatus = append(rStatus, eachReviewer)
+
+	for _, elem := range reviewers {
+		review := github.PullRequestReview{}
+		review.User = elem
+
+		// Any reviews are later than this assignment, so put it farthest back.
+		assignDate := time.Unix(0, 0)
+		review.SubmittedAt = &assignDate
+
+		// Assume their review is pending until further notice.
+		status := "PENDING"
+		review.State = &status
+
+		reviews[elem.GetLogin()] = []github.PullRequestReview{review}
 	}
-	return rStatus
-}
 
-func getReviewStatesForPR(repo string, number int) []ReviewState {
-	rReviews = nil
-	// List all reviewers to an assigned PR # in a repo.
-	rList, _, err := buzzClient.PullRequests.ListReviews(ctx, "minio", repo, number, nil)
+	// List all who voluntarily gave a review.
+	voluntaryReviews, _, err := buzzClient.PullRequests.ListReviews(ctx, "minio", repo, number, nil)
 	if err != nil {
-		// We should not exit on error in this case.
-		fmt.Println("Unable to get Reviewers for %s", number)
+		return err, nil
 	}
-	for _, elem := range rList {
-		eachReview := ReviewState{}
-		eachReview.User.Login = elem.User.GetLogin()
-		eachReview.State = elem.GetState()
-		rReviews = append(rReviews, eachReview)
+
+	for _, elem := range voluntaryReviews {
+		// If a review is dismissed, change to PENDING. Ignore comments.
+		if elem.GetState() == "DISMISSED" {
```

Comment:
This can be converged into an `switch` case .. with duplicate code avoided. 

```
switch elem.GetState() {
case "COMMENTED":
      continue
default:
  if _, ok := reviews[elem.User.GetLogin()]; ok {
  	reviews[elem.User.GetLogin()] = append(reviews[elem.User.GetLogin()], *elem)
  } else {
        reviews[elem.User.GetLogin()] = []github.PullRequestReview{*elem}
}
```